pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,227 @@
+  - tools
+  - code generation
+cover: /img/posts/jonaslagoni-miniseries-part1/blog-miniseries-cover.webp
+authors:
+  - name: Jonas Lagoni
+    photo: /img/avatars/jonaslagoni.webp
+    link: https://github.com/jonaslagoni
+    byline: AsyncAPI Maintainer
+featured: true
+---
+
+How can you utilize code generation to speed up the development process and only focus on what is important, the business logic? In this mini-series we will explore the ways AsyncAPI and code generation can work hand in hand beyond generating documentation. 
+
+Structure of the miniseries:
+* **Part 1: Designing the API's with AsyncAPI**
+* Part 2: Implementing the applications using code generation
+* Part 3: Black-box testing the applications using code generation
+* Part 4: Introducing new changes when using code generation
+* Part 5: The path to 1 billion players - Scaling the applications and finding bottlenecks with code generation
+
+**Don't see this blog post series as anything other then an example workflow, this is purely how I do it with my applications and how I use AsyncAPI and it's tooling to my advantage. Use this as an inspiration to finding an approach that works for you.**
+
+# Backstory
+
+Back in 2019 when I started contributing to the tooling of AsyncAPI, I was still in university studying for a masters in software engineering, and had at that point been a student developer at a company called [EURISCO](http://eurisco.dk/), for about 3 years. Beside that I have always had side projects that I worked on in my spare time, and it was one of these side projects that sparked my need for AsyncAPI. 
+
+My side project at that time was a [Rust](https://rust.facepunch.com/) game server plugin which collected in game events, such as when a player farms resources, kills another player, loots a container, etc, and send them to a backend. Later these could be extracted by an API to display the players progression and detailed account on what you did on the game server. 
+
+Initially I used OpenAPI to describe the API and the great thing was their tooling allowed me generate clients and servers in different languages which really speed up the implementation process.
+
+I soon encountered a use case that required me to push data to the game server, and solving this with REST was possible, but cumbersome. So I started exploring different alternatives, in terms of event driven architecture, however, none could be described using OpenAPI or offered tooling, so I had to find alternatives. 
+
+That was when I vaguely remembered a meeting where AsyncAPI was mentioned. Around that time, at work, we began to switch from a custom socket protocol to [NATS](https://nats.io/), and at the same time they spend some time figuring out how to mainstream the process and document the API's, and this is where they found and adopted AsyncAPI. So I started to use AsyncAPI in my project, and sparked my first ever encounter with open source, but that is a story for another time, maybe.
```

Comment:
I lost track here, I think you are referring to EURISCO again, but I'm not 100% sure, so be explicit here

Reply:
Good point added 👍